specification/resources/resource-manager/Microsoft.Authorization/preview/2023-03-01-preview/privateLinks.json

```diff
@@ -0,0 +1,834 @@
+          "readOnly": true,
+          "type": "string",
+          "description": "The rmpl Name."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The operation type."
+        },
+        "location": {
+          "type": "string",
+          "description": "the region of the rmpl"
+        },
+        "tags": {
+          "description": "The resource tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "ResourceManagementPrivateLinkLocation": {
+      "type": "object",
+      "properties": {
+        "location": {
+          "type": "string",
+          "description": "the region to create private link association."
+        }
+      }
+    },
+    "ResourceManagementPrivateLinkPatch": {
+      "type": "object",
+      "properties": {
+        "tags": {
+          "description": "Resource tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string",
+            "x-nullable": true
+          }
+        }
+      }
+    },
+    "ResourceManagementPrivateLinkEndpointConnections": {
+      "type": "object",
+      "properties": {
+        "privateEndpointConnections": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The private endpoint connections."
+        }
+      }
+    },
+    "ResourceManagementPrivateLinkListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ResourceManagementPrivateLink"
+          },
+          "description": "An array of resource management private links."
+        }
+      }
+    },
+    "CloudError": {
+      "x-ms-external": true,
+      "type": "object",
+      "properties": {
+        "error": {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/ErrorResponse"
+        }
+      },
+      "description": "An error response for a resource management request."
+    }
+  },
+  "parameters": {
+    "ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "The API version to use for this operation.",
+      "x-ms-parameter-location": "client"
+    },
+    "GroupIdParameter": {
+      "name": "groupId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The management group ID.",
+      "x-ms-parameter-location": "method",
+      "minLength": 1,
+      "maxLength": 90
+    },
+    "SubscriptionIdParameter": {
+      "name": "subscriptionId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The ID of the target subscription.",
+      "x-ms-parameter-location": "method"
+    },
```

Comment:
These API Version, SUbscription Id etc. properties exist in common types: [https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v5). Instead of defining here, please use from common types.

Reply:
+1, any reason not to?

Reply:
I tried to move to v5 commonTypes in the previous commit. Had to revert it back because it is a breaking change for cross version.  Note this is just copied from version 2020-05-01. The only new API added in this PR is POST enablePublicNetworkAccess. Revert commit: https://github.com/Azure/azure-rest-api-specs/commit/124d2c35767141e373b3ecb256fcb15e359d1eaf